crates/bevy_ecs/src/bundle.rs

```diff
@@ -106,21 +108,23 @@ macro_rules! tuple_impl {
 
             #[allow(unused_variables, unused_mut)]
             #[allow(clippy::unused_unit)]
-            unsafe fn from_components(mut func: impl FnMut() -> *mut u8) -> Self {
+            unsafe fn from_components<T, F>(ctx: &mut T, mut func: F) -> Self
```

Comment:
Does this mean that the bundle can only live as long as `ctx`? If we destroy the world, will this cause UB?

Reply:
Can we just give the `Bundle` trait a `'world` lifetime?

Reply:
I would expect `ctx` to exclusively borrow the world?

How would we destroy the world in that case?

Reply:
No it doesn't mean bundle can only live as long as `ctx` and `Bundle` doesnt need a `'world` lifetime. This function is used when removing a bundle from world so we are taking data out of `ctx` hence the `OwningPtr<'_>` and moving it into `Self` to construct the bundle. The only lifetimes involved are for the temporary borrow on world while we are removing the components to construct our bundle from.